server/src/lib.rs

```diff
@@ -224,17 +227,72 @@ async fn mirror(
         };
 
     let request = request.expect("geth responded, so body must have been JSON");
-    let op_move_response = moved_engine_api::request::handle(request.clone(), state_channel).await;
+    let op_move_response =
+        moved_engine_api::request::handle(request.clone(), state_channel.clone()).await;
     let log = MirrorLog {
         request: &request,
         geth_response: &parsed_geth_response,
         op_move_response: &op_move_response,
         port,
     };
     println!("{}", serde_json::to_string_pretty(&log).unwrap());
-    // TODO: use op_move_response
-    let body = hyper::Body::from(geth_response_bytes);
-    Ok(warp::reply::Response::from_parts(geth_response_parts, body))
+
+    // TODO: this is a hack because we currently can't compute the genesis
```

Comment:
We can keep these operations in a separate file, so that this one time operation doesn't take so much code space in here. But it's also Ok to keep it here.

Reply:
Factored out in https://github.com/MovedNetwork/op-move/pull/168/commits/61a1da6b6bc589a7670fad19a9ebab2930bc049d